tests/functional/configs/test_contract_configs.py

```diff
@@ -367,22 +396,49 @@ class TestModelLevelConstraintsErrorMessages:
     @pytest.fixture(scope="class")
     def models(self):
         return {
-            "my_model.sql": my_ephemeral_model_sql,
-            "constraints_schema.yml": model_schema_errors_yml,
+            "my_model.py": my_model_python_error,
+            "constraints_schema.yml": model_schema_yml,
         }
 
     def test__config_errors(self, project):
         with pytest.raises(ParsingError) as err_info:
             run_dbt(["run"], expect_pass=False)
 
         exc_str = " ".join(str(err_info.value).split())
-        expected_materialization_error = "Only table, view, and incremental materializations are supported for constraints, but found 'ephemeral'"
+        expected_materialization_error = "Language Error: Expected 'sql' but found 'python'"
         assert expected_materialization_error in str(exc_str)
         # This is a compile time error and we won't get here because the materialization check is parse time
         expected_empty_data_type_error = "Columns with `data_type` Blank/Null not allowed on contracted models. Columns Blank/Null: ['date_day']"
         assert expected_empty_data_type_error not in str(exc_str)
 
 
+class TestModelLevelConstraintsWarningMessages:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "my_model.sql": my_view_model_sql,
+            "constraints_schema.yml": model_schema_yml,
+        }
+
+    def test__config_warning(self, project):
+        _, log_output = run_dbt_and_capture(["run"])
+
+        expected_materialization_warning = (
+            "Constraint types are not supported for view materializations"
+        )
+        assert expected_materialization_warning in str(log_output)
+
+        # change to not show warnings, message should not be in logs
+        models_dir = os.path.join(project.project_root, "models")
+        write_file(model_schema_ignore_unsupported_yml, models_dir, "constraints_schema.yml")
+        _, log_output = run_dbt_and_capture(["run"])
+
+        expected_materialization_warning = (
+            "Constraint types are not supported for ephemeral materializations."
```

Comment:
should this be 'view' instead of 'ephemeral'?

Reply:
Great catch!

Reply:
Should this be an adapter zone test?

Reply:
This specifically is not adapter specific.  There are tests on constraints in the adapter zone already.